File: notification_service/python/controllers/notificationController.py
from flask import jsonify, request
from pymongo import MongoClient
from bson.objectid import ObjectId
from datetime import datetime

client = MongoClient('mongodb://localhost:27017/')
db = client['social_media_platform']
notifications = db['notifications']

def get_notifications():
    all_notifications = list(notifications.find())
    return jsonify(all_notifications)

def create_notification(req):
    data = req.get_json()
    notification = {
        "userId": ObjectId(data['userId']),
        "message": data['message'],
        "createdAt": datetime.utcnow()
    }
    notifications.insert_one(notification)
    return jsonify(notification), 201

File: notification_service/nodejs/controllers/notificationController.js
const { MongoClient, ObjectId } = require('mongodb');
const url = 'mongodb://localhost:27017';
const dbName = 'social_media_platform';

const getNotifications = async (req, res) => {
  const client = new MongoClient(url);
  try {
    await client.connect();
    const db = client.db(dbName);
    const notifications = await db.collection('notifications').find().toArray();
    res.json(notifications);
  } finally {
    client.close();
  }
};

const createNotification = async (req, res) => {
  const client = new MongoClient(url);
  const { userId, message } = req.body;
  try {
    await client.connect();
    const db = client.db(dbName);
    const result = await db.collection('notifications').insertOne({
      userId: ObjectId(userId),
      message,
      createdAt: new Date()
    });
    res.status(201).json(result.ops[0]);
  } finally {
    client.close();
  }
};

module.exports = {
  getNotifications,
  createNotification
};
